connector/subscription_manager.go

```diff
@@ -34,22 +34,26 @@ type subscriptionmanager struct {
 	conn atomic.Value
 }
 
-func (m *subscriptionmanager) Add(topic string) {
+func (m *subscriptionmanager) Add(topic string) bool {
 	if _, ok := m.topics.LoadOrStore(topic, true); !ok {
 		if connRef := m.conn.Load(); connRef != nil {
 			conn := connRef.(*MQTTConnection)
 			conn.subscribe(nil, QosAtMostOnce, topic)
 		}
+		return true
```

Comment:
It's possible to have stored topic but not registered in case of connRef == nil.